index.js

```diff
@@ -5,18 +5,44 @@
  */
 
 const rescue = f => source => (start, sink) => {
-  start === 0 && source(start, (t, d) => {
-    if (t === 2 && d != null) {
-      try {
-        const newSource = f(d);
-        newSource(start, sink);
-      } catch (e) {
-        sink(2, e);
+  if (start !== 0) return;
+
+  let rescued = false;
+  let sourceTalkback;
+
+  const talkback = (t, d) => {
```

Comment:
why is this wrapper necessary? :-) I think sourceTalkback is enough

Reply:
because we swap `sourceTalkback` after rescuing for the newSource's talkback, but we have already passed `talkback` to the sink and we cant force it to swap references, so we have to do it on our side here